src/main/java/com/autotune/analyzer/services/UpdateRecommendations.java

```diff
@@ -81,16 +82,18 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         LOGGER.debug("UpdateRecommendations API request count: {}", calCount);
         String statusValue = "failure";
         Timer.Sample timerBUpdateRecommendations = Timer.start(MetricsConfig.meterRegistry());
-        try {
-            // Set the character encoding of the request to UTF-8
-            request.setCharacterEncoding(CHARACTER_ENCODING);
-            // Get the values from the request parameters
-            String experiment_name = request.getParameter(KruizeConstants.JSONKeys.EXPERIMENT_NAME);
-            String intervalEndTimeStr = request.getParameter(KruizeConstants.JSONKeys.INTERVAL_END_TIME);
+        // Set the character encoding of the request to UTF-8
+        request.setCharacterEncoding(CHARACTER_ENCODING);
+        // Get the values from the request parameters
+        String experiment_name = request.getParameter(KruizeConstants.JSONKeys.EXPERIMENT_NAME);
+        String intervalEndTimeStr = request.getParameter(KruizeConstants.JSONKeys.INTERVAL_END_TIME);
 
-            String intervalStartTimeStr = request.getParameter(KruizeConstants.JSONKeys.INTERVAL_START_TIME);
-            Timestamp interval_end_time;
-            Timestamp interval_start_time = null;
+        String intervalStartTimeStr = request.getParameter(KruizeConstants.JSONKeys.INTERVAL_START_TIME);
+        Timestamp interval_end_time;
+        Timestamp interval_start_time = null;
+        if (KruizeDeploymentInfo.logAllHttpReqAndResp)
```

Comment:
We ought to consolidate logging for this specific area instead of having  flag across all services. Utilizing an HTTP filter would be more appropriate; we can explore this in a separate PR